### PR TITLE
8332245: C2: missing record_for_ign() call in GraphKit::must_be_not_null()

### PR DESCRIPTION
**Pull request by rwestrel**
The `If` node that's created by `GraphKit::must_be_not_null()` is not
enqueued for igvn when it's created. The test case shows it prevents 2
identical tests from commoning when the first igvn executes. This is a
minor issue I noticed whule working on something else.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332245](https://bugs.openjdk.org/browse/JDK-8332245): C2: missing record_for_ign() call in GraphKit::must_be_not_null() (**Bug** - P5)

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [48fb906c](https://git.openjdk.org/jdk/pull/19233/files/48fb906c53dd1a6ac8783678f23056288f2b53d8)
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [48fb906c](https://git.openjdk.org/jdk/pull/19233/files/48fb906c53dd1a6ac8783678f23056288f2b53d8)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19233/head:pull/19233` \
`$ git checkout pull/19233`

Update a local copy of the PR: \
`$ git checkout pull/19233` \
`$ git pull https://git.openjdk.org/jdk.git pull/19233/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19233`

View PR using the GUI difftool: \
`$ git pr show -t 19233`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19233.diff">https://git.openjdk.org/jdk/pull/19233.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19233#issuecomment-2110643452)

### CODE REVIEW

File: src/hotspot/share/opto/graphKit.cpp [hunk 1]
```diff
@@ -1468,6 +1468,7 @@ Node* GraphKit::must_be_not_null(Node* value, bool do_replace_in_map) {
   Node* opaq = _gvn.transform(new Opaque4Node(C, tst, intcon(1)));
   IfNode *iff = new IfNode(control(), opaq, PROB_MAX, COUNT_UNKNOWN);
   _gvn.set_type(iff, iff->Value(&_gvn));
+  if (!tst->is_Con())  record_for_igvn(iff);
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/graphKit.cpp:1471]:
```suggestion
  if (!tst->is_Con()) {
    record_for_igvn(iff);
  }
```

**Review by TobiHartmann — APPROVED** [review]
Looks good to me otherwise.

**Review by chhagedorn — APPROVED** [review]
Looks good to me, too.

**Comment by rwestrel** [comment]
@TobiHartmann @chhagedorn thanks for the reviews

**Pull request closed without merging**